### ISSUE TITLE
shiny application breaks with release 1.8.1

### ISSUE THREAD

**Issue by martinamorris** [rstudio/shiny#4099]
We have an older shiny app, [statnetWeb](https://github.com/statnet/statnetWeb) that broke between releases 1.8.0 and 1.8.1.   

The page rendering no longer works properly for a page that uses direct calls to `conditionalPanel()` and `uiOutput()`, and since your [1.8.1 release notes](https://github.com/rstudio/shiny/releases/tag/v1.8.1) explicitly state these have changed behavior that might break apps that call them directly, I'm thinking this might be our problem.

I'm not sure how to create a self-contained reprex for this, and I'm not an expert in coding shiny apps.  But I can show the problem.

Desired behavior can be observed currently on the app hosted on the shiny server:  https://statnet.shinyapps.io/statnetWeb/

Select the "Data" tab up at the top, and work through the options.  It will look like this:

![image](https://github.com/user-attachments/assets/f15f97cc-0664-4e0c-992a-bff0d507a286)

1. The screen updates interactively with different input options as you choose a "File type" -- compare "built-in network" to any of the other choices.  
2. Once you choose a built-in network, the network is loaded and the summaries appear on the right tab, and you can continue working with it.

Using shiny 1.8.1, and running the app locally, this is what the "Data" page looks like now -- note the difference with the hosted app:

![image](https://github.com/user-attachments/assets/799de75d-97d3-45f0-82c9-4fbaab3e5f8e)

1. Nothing changes interactively now with the File type selection
2. It doesn't load the data when you "Choose a network"

The code for this page of the app is found [here](https://github.com/statnet/statnetWeb/blob/4f90fdc471a4c6d19acb325e615e2a4a03b4fdf7/inst/shiny/statnetWeb/ui.R#L152-L474)

It calls `conditionalPanel()` directly and repeatedly.  

I've isolated the shiny release change as the cause by reinstalling shiny 1.8.0 from the CRAN archive and running that version locally.  The behavior of 1.8.0 matches the desired behavior, as seen on the hosted app.

Hopefully this is enough to identify what the issue is, and whether there's a simple fix for our code to update to shiny 1.8.1.

TIA!  Shiny is awesome :)

**Comment by gadenbuie** [comment]
Thanks for opening this issue, @martinamorris. I can definitely see differences between the deployed app and the app with updated Shiny dependencies.

First, I'd like to clarify the note in the changelog. The 1.8.1 release notes mention

> Both `conditionalPanel()` and `uiOutput()` are now styled with `display: contents` by default in Shiny apps that use Bootstrap 5. ...  it may break apps that applied styles directly to the container elements created by these two functions.

I should have probably written "may break the appearance of apps". The changes described in that news bullet would only affect the appearance of `conditionalPanel()` outputs, but not the overall behavior. In other words, you'd still see the conditional panel contents and the conditional panel will still work as expected.

In summary, I don't think that change is at the root of changes in app behavior between these shiny versions.

I think it's much more likely that the issues are stemming from the app loading jQuery 1.11 [in these lines](https://github.com/statnet/statnetWeb/blob/4f90fdc471a4c6d19acb325e615e2a4a03b4fdf7/inst/shiny/statnetWeb/ui.R#L174-L178). That version of jQuery may have been in line with the version used by Shiny when the app was written, but it's quite out of date now.

If I comment out those lines, the core Shiny logic of the app seems to work as expected.

**Comment by martinamorris** [comment]
Wow, thank you so much for both the quick response, and for taking the time to find the problem.  Core shiny logic works now, other package related stuff not so much :-/

This is a legacy application, and so much has changed since it was written (10 years ago!).  It really needs a full rewrite at this point.

Again, many many thx for your help.
